docs/chapter2-payload-image-format.rst

```diff
@@ -361,7 +361,7 @@ included to be used later by the main image after the main image is executed.
     			data-offset = <...data…>;
     			data-size = <...data…>;
     			reloc-start = <start offset of reloc table within data>;
-    			entry-start = <0x121b10>;
+    			entry = <0x121b10>;
```

Comment:
If this is supposed to be an offset from the load address, shouldn't it be 0x1b10 ?

Reply:
yes, good catch!

Reply:
No, that's not how it's used at the moment. "entry-start" (now, "entry") is an absolute address, not relative one. See below:

```
        tianocore {
            description = "Uefi Universal Payload";
            project = "tianocore";
            arch = "x86";
            type = "flat-binary";
            producer = "intel";
            data-offset = <0x00001000>;
            data-size = <0x00016000>;
            reloc-start = <0x00012000>;
            entry-start = <0x00000000 0x00808e50>;
            load = <0x00000000 0x00801000>;
        };
```

However, perhaps `entry` *should* be an offset. After all, if the Platform Init performs relocation, it's no longer correct

Reply:
For reference, this is what the FIT spec says about it. See section 2.3.2:
> entry
>
>     entry point address, address size is determined by ‘#address-cells’ property of the root node. Mandatory for types: “firmware”, and “kernel”.
>
> load
>
>     load address, address size is determined by ‘#address-cells’ property of the root node. Mandatory for types: “firmware”, and “kernel”.

It makes no mention of being relative, so we might have to continue with it being absolute, even if it doesn't make sense. Does someone have another (non-UPL) FIT to confirm this against?

Reply:
Yes, it is absolute in FIT itself.

I think absolute is fine since if we relocate we can adjust that too.

Reply:
thanks for clarifying Benjamin!

Reply:
> Yes, it is absolute in FIT itself.

@sjg20 Over here, it's an absolute address in the host memory space. With the image I sent above, if you copy it in to `0x00801000`, then you start it by calling to `0x00808e50`. It seemed possible there's a misunderstanding here? If not, we can mark this as resolved.

Reply:
Anyways, I agree it's not a big deal: if someone is relocating it, they can shift entry while they're shifting load.

Reply:
Ok will update this to absolute address, thanks.